nautobot_floor_plan/utils/general.py

```diff
@@ -0,0 +1,146 @@
+            prefix = label[:i]
+            numbers = label[i:]
+            break
+
+    return prefix, numbers
+
+
+def letter_conversion(converted_location):
+    """Returns letter conversion and handles wrap around."""
+    # Set wrap around value of ZZZ
+    total_cells = 18278  # Total cells for AAA-ZZZ
+    # Adjust for wrap-around when working with letters A or ZZZ
+    if converted_location < 1:
+        converted_location = total_cells + converted_location
+    elif converted_location > total_cells:
+        converted_location -= total_cells
+    return grid_number_to_letter(converted_location)
+
+
+def axis_init_label_conversion(axis_origin, axis_location, step, is_letters):
+    """Convert an axis location to its label based on the origin, step, and label type."""
+    try:
+        if is_letters:
+            axis_location = grid_letter_to_number(axis_location)
+
+        converted_location = axis_origin + (int(axis_location) - int(axis_origin)) * step
+
+        if is_letters:
+            return letter_conversion(converted_location)
+
+        return converted_location
+    except ValueError as e:
+        raise ValueError(
+            f"Error in axis conversion: axis_origin={axis_origin}, "
+            f"axis_location={axis_location}, step={step}, is_letters={is_letters}"
+        ) from e
+
+
+def axis_clean_label_conversion(axis_origin, axis_label, step, is_letters, custom_ranges=None):
+    """Returns the correct database label position."""
+    # First check if we have custom ranges that apply
+    if custom_ranges:
+        for custom_range in custom_ranges:
+            start = custom_range["start"]
+            end = custom_range["end"]
+            # If our label falls within this custom range, use it
+            if start <= axis_label <= end:
+                return axis_label  # Return the original label as-is
+
+    # If no custom range applies, use the default conversion logic
+    total_cells = 18278
+    # Convert letters to numbers if needed
+    if is_letters:
+        axis_label = grid_letter_to_number(axis_label)
+
+    # Reverse the init conversion logic to determine the numeric position
+    position_difference = int(axis_label) - int(axis_origin)
+
+    if step < 0:
+        # Adjust for wrap-around when working with letters A or ZZZ
+        if int(axis_label) > int(axis_origin):
+            position_difference -= total_cells
+    else:
+        if int(axis_label) < int(axis_origin):
+            position_difference += total_cells
+
+    # Calculate the original location using the step
+    original_location = axis_origin + (position_difference // step)
+
+    # Ensure original location stays within bounds for letters
+    if is_letters:
+        if original_location < 1:
+            original_location += total_cells
+        elif original_location > total_cells:
+            original_location -= total_cells
+    return str(original_location)
+
+
+def calculate_letter_range_size(custom_range, converter):
```

Comment:
looks like dead code? I couldn't find it's reference in-use though theres a similar method in `custom_validators.py` -> `_calculate_letter_range_size`